staticd/static_zebra.c

```diff
@@ -224,20 +280,48 @@ void static_zebra_nht_register(struct static_route *si, bool reg)
 		p.family = AF_INET;
 		p.prefixlen = IPV4_MAX_BITLEN;
 		p.u.prefix4 = si->addr.ipv4;
+		afi = AFI_IP;
 		break;
 	case STATIC_IPV6_GATEWAY:
 	case STATIC_IPV6_GATEWAY_IFNAME:
 		p.family = AF_INET6;
 		p.prefixlen = IPV6_MAX_BITLEN;
 		p.u.prefix6 = si->addr.ipv6;
+		afi = AFI_IP6;
 		break;
 	}
 
+	memset(&lookup, 0, sizeof(lookup));
+	lookup.nh = &p;
+
+	si->nh_registered = reg;
+
+	if (reg) {
+		nhtd = hash_get(static_nht_hash, &lookup,
+				static_nht_hash_alloc);
+		nhtd->refcount++;
+
+		if (nhtd->refcount > 1) {
+			static_nht_update(nhtd->nh, nhtd->nh_num,
```

Comment:
Just curious about this added clause: is there a condition where registering interest in a tracked nexthop would change the status of the nexthop?

Reply:
no, they should be independent.  Registering a nexthop tells zebra to just let you know when something changes that is it.